### ISSUE TITLE
Flask + uWSGI - traces_sampler is returning different transaction then which was sent to the Sentry

### ISSUE THREAD

**Issue by ldath** [getsentry/sentry-python#913]
Hello, 

I installed version 0.19.2 hoping that will solve my problem with Kubernetes probe, unfortunately result was really strange.

With this code:
```
def traces_sampler_callback(traces_sampler):
    with open('/tmp/sampler.txt', 'w') as f:
        f.write(json.dumps(traces_sampler, indent=4, sort_keys=True, default=str))
    return 0.01


sentry_sdk.init(
    dsn=YamlConfig().get_sentry_dsn(),
    release="web-pages-admin@{}".format(os.environ.get('APP_VERSION', 'latest')),
    integrations=[FlaskIntegration(), SqlalchemyIntegration()],
    # traces_sample_rate=0.01,
    traces_sampler=traces_sampler_callback
)
```
for the request to the probe in the file I have this:
```
{
    "parent_sampled": null,
    "transaction_context": {
        "description": null,
        "name": "generic WSGI request",
        "op": "http.server",
        "parent_sampled": null,
        "parent_span_id": null,
        "same_process_as_parent": false,
        "sampled": null,
        "span_id": "b6523eeda1936e8f",
        "start_timestamp": "2020-11-06 18:43:30.462145",
        "timestamp": null,
        "trace_id": "f7d099e9e1f64d499cdce1ab9cf24a2b"
    }
}
```

and when I'm looking at the view:
```
@main.route('/probe', methods=['GET', 'HEAD'])
def probe():
    with configure_scope() as scope:
        with open('/tmp/transaction.txt', 'w') as f:
            f.write(json.dumps(scope.transaction.to_json(), indent=4, sort_keys=True, default=str))
    return jsonify(status="success", message="I'm alive")
```
my result is looking like the data I see in the sentry.io
```
{
    "description": null,
    "name": "main.probe",
    "op": "http.server",
    "parent_sampled": null,
    "parent_span_id": null,
    "same_process_as_parent": false,
    "sampled": false,
    "span_id": "8d02796c52128ec0",
    "start_timestamp": "2020-11-06 18:43:30.386167",
    "timestamp": null,
    "trace_id": "9259e3e066514a1eb73b5e92a3237486"
}
```

As you can see - `span_id` and `trace_id` is different and totally not useful in case of my traces_sampler callback. If will be this I see in view than I can filter my probes but `"name": "generic WSGI request",` is not useful for me and I should have view trace with name "main.probe" which will be useful. 

Why we have this one which looks like uWSGI instead of the transaction which is really used and I need to filter?

**Comment by untitaker** [comment]
>"name": "generic WSGI request", is not useful for me

Please refer to https://github.com/getsentry/sentry-python/issues/775#issuecomment-721283037 about this problem. Basically the issue is that Flask routing happens halfway through the transaction which is too late for sampling decisions. Going forward you will at least have raw request url available in the context which should help.

>span_id and trace_id is different

This is really weird. The timestamp is also very close together. Could it however be that those transactions are from two different HTTP requests?

**Comment by ldath** [comment]
> > "name": "generic WSGI request", is not useful for me
> 
> Please refer to [#775 (comment)](https://github.com/getsentry/sentry-python/issues/775#issuecomment-721283037) about this problem. Basically the issue is that Flask routing happens halfway through the transaction which is too late for sampling decisions. Going forward you will at least have raw request url available in the context which should help.

When this will be available? Next version after 0.19.2 ?

> 
> > span_id and trace_id is different
> 
> This is really weird. The timestamp is also very close together. Could it however be that those transactions are from two different HTTP requests?

Exactly the same request - that is why this looks very strange for me. But it have sense as when I'm checking in the view which is sending response I see exactly the same name of the transaction like I see in the sentry.io . 
Of course it is possible that doing this flask routing transaction change name but then why ID is different?

**Comment by untitaker** [comment]
>When this will be available? Next version after 0.19.2 ?

yes exactly

> Of course it is possible that doing this flask routing transaction change name but then why ID is different?

I would create a separate issue for this, I think this needs detailed investigation but I don't think every subscriber to this thread would want to be notified about this.

**Comment by ldath** [comment]
OK, now this is clear and can be easlly checked. When I'm using  browser:

```
{
    "parent_sampled": null,
    "transaction_context": {
        "description": null,
        "name": "generic WSGI request",
        "op": "http.server",
        "parent_sampled": null,
        "parent_span_id": null,
        "same_process_as_parent": false,
        "sampled": null,
        "span_id": "b522e96060151f38",
        "start_timestamp": "2020-11-16 21:09:17.618610",
        "timestamp": null,
        "trace_id": "0ae06ebc2ddc487aa9627587e798be4a"
    },
    "wsgi_environ": {
        "HTTP_ACCEPT": "image/avif,image/webp,image/apng,image/*,*/*;q=0.8",
        "HTTP_ACCEPT_ENCODING": "gzip, deflate, br",
        "HTTP_ACCEPT_LANGUAGE": "pl-PL,pl;q=0.9,en-US;q=0.8,en;q=0.7",
        "HTTP_CONNECTION": "close",
        "HTTP_COOKIE": "session=.eJwlzs0KAjEMBOB36dlD2qRtui-zpPlBERR29SS-uwVvM8Mcvk_a4_DzmrbX8fZL2m-WtsRSJfceOKGwteGI2lDMutWYhkVsYjPuYjBDRbFPxjD0RgTSMdSUSbQOo27Do7Cg4OCGGDUgtwBurcKYwZUKDgK1cI-gnGtakPfpx1-TV9XziP31vPtjDZjXzReOtZMIURiBrcRuWAtkXdAYmr4_vWtBMA.X7LqXA.olqdJohR43ZTk8LWLWpd1w__dg0",
        "HTTP_HOST": "dev-admin2...",
        "HTTP_REFERER": "https://dev-admin2.../probe",
        "HTTP_SEC_FETCH_DEST": "image",
        "HTTP_SEC_FETCH_MODE": "no-cors",
        "HTTP_SEC_FETCH_SITE": "same-origin",
        "HTTP_USER_AGENT": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36",
        "HTTP_X_FORWARDED_FOR": "172.26.0.1",
        "HTTP_X_FORWARDED_PROTO": "https",
        "HTTP_X_FORWARDED_SSL": "on",
        "HTTP_X_FRAME_OPTIONS": "SAMEORIGIN",
        "HTTP_X_OUTSIDE_URL": "https://dev-admin2...:443",
        "HTTP_X_REAL_IP": "172.26.0.1",
        "PATH_INFO": "/favicon.ico",
        "QUERY_STRING": "",
        "REMOTE_ADDR": "172.26.0.13",
        "REMOTE_PORT": "31920",
        "REQUEST_METHOD": "GET",
        "REQUEST_URI": "/favicon.ico",
        "SCRIPT_NAME": "",
        "SERVER_NAME": "d9681c42ad85",
        "SERVER_PORT": "5000",
        "SERVER_PROTOCOL": "HTTP/1.0",
        "UWSGI_ROUTER": "http",
        "uwsgi.node": "d9681c42ad85",
        "uwsgi.version": "2.0.17",
        "wsgi.errors": "<_io.TextIOWrapper name=2 mode='w' encoding='UTF-8'>",
        "wsgi.file_wrapper": "<built-in function uwsgi_sendfile>",
        "wsgi.input": "<uwsgi._Input object at 0x7f13ff541690>",
        "wsgi.multiprocess": true,
        "wsgi.multithread": false,
        "wsgi.run_once": false,
        "wsgi.url_scheme": "https",
        "wsgi.version": [
            1,
            0
        ]
    }
}
```
and Transaction is different because browser after taking JSON data is accessing `favicon.ico` which I'm also managing with Flask.

When I'm using curl only one request is going and the `trace_id` is the same. I just forgot that browsers always are going to take this `favicon.ico` file.
